docs/src/components/landing/DemoVideo.js

```diff
@@ -155,6 +163,44 @@ export default function DemoVideo() {
     setIsMuted(videoRef.current.muted);
   };
 
+  const milestonesComplete = React.useRef(new Set([]));
+
+  const handleTimeUpdate = () => {
+    if (typeof window === 'undefined' || !window.gtag || !videoRef.current) {
+      return;
+    }
+    const videoElement = videoRef.current;
+
+    const videoProgress = (videoElement.currentTime / videoElement.duration) * 100;
+
+    if (videoProgress >= 25 && videoProgress < 50 && !milestonesComplete.current.has(25)) {
+      window.gtag('event', 'toolpad_video', {
+        event_category: 'toolpad_landing',
+        event_label: `25% Complete`,
+      });
+      milestonesComplete.current.add(25);
+    } else if (videoProgress >= 50 && videoProgress < 75 && !milestonesComplete.current.has(50)) {
+      window.gtag('event', 'toolpad_video', {
+        event_category: 'toolpad_landing',
+        event_label: `50% Complete`,
+      });
+      milestonesComplete.current.add(50);
+    } else if (videoProgress >= 75 && videoProgress < 100 && !milestonesComplete.current.has(75)) {
+      window.gtag('event', 'toolpad_video', {
+        event_category: 'toolpad_landing',
+        event_label: `75% Complete`,
+      });
+      milestonesComplete.current.add(75);
+    } else if (videoProgress === 100 && !milestonesComplete.current.has(100)) {
+      window.gtag('event', 'toolpad_video', {
+        event_category: 'toolpad_landing',
+        event_label: `100% Complete`,
+      });
+      milestonesComplete.current.add(100);
+      milestonesComplete.current.clear();
```

Comment:
Why?

Reply:
It would, that was the intent since I thought we would want to count that as a separate "watch"